src/hotspot/share/opto/escape.cpp

```diff
@@ -4101,6 +4106,20 @@ void ConnectionGraph::split_unique_types(GrowableArray<Node *>  &alloc_worklist,
         nmm->set_memory_at(ni, result);
       }
     }
+
+    // If we have crossed the 3/4 point of max node limit it's too risky
+    // to continue with EA/SR because we might hit the max node limit.
+    if (_compile->live_nodes() >= _compile->max_node_limit() * 0.75) {
+      if (_compile->do_reduce_allocation_merges()) {
+        _compile->record_failure(C2Compiler::retry_no_reduce_allocation_merges());
+      } else if (_invocation > 0) {
+        _compile->record_failure(C2Compiler::retry_no_iterative_escape_analysis());
+      } else {
+        _compile->record_failure(C2Compiler::retry_no_escape_analysis());
+      }
+      return;
+    }
+
```

Comment:
I don't see the changes to this file that should come from https://github.com/openjdk/jdk/pull/19147/files (e.g., changes to split_castpp_load_through_phi). Am I overlooking them, or did they not need to be backported?

Reply:
Sorry, I put the wrong link in the PR description. Instead of 19147 I meant to place [#18851](https://github.com/openjdk/jdk/pull/18851/files),